main.js

```diff
@@ -11,31 +11,41 @@ function generateNumbers() {
     return allNumbers;
 }
 
-function displayNumbers(allNumbers) {
-    let numbersOnScreen = allNumbers.length;
-
+function displayNumbers() {
+    const randomNumbers = generateNumbers();
+    const sortedNumbers = randomNumbers.slice(0).sort();
+    let numbersOnScreen = randomNumbers.length;
      // using a for loop instead of forEach as the array.pop changes the length over each iteration
      for (let i = 0; i < numbersOnScreen; i++) {
-        let randNumber = allNumbers.pop();
+        let randNumber = randomNumbers.pop();
         let number = document.createElement('div');
 
         number.textContent = randNumber;
         number.setAttribute('class', 'numberBox');
 
          document.querySelector('#play_area').appendChild(number);
     }
-
+    let selectedOrder = [];
     document.querySelectorAll('.numberBox').forEach((element) => {
         element.addEventListener('click', () => {
+
+            const { textContent, classList } = element;
+            const value = parseInt(textContent);
+            if (sortedNumbers[selectedOrder.length] === value) {
+                selectedOrder.push(value);
+                classList.add('chosen_number');
+            }
+
+            if (selectedOrder === sortedNumbers) {
+// all selected correctly
+            }
         });
     });
 }
 
 document.querySelector('#start').addEventListener('click', () => {
     document.querySelector('#splash').style.display = 'none';
     document.querySelector('#game').style.display = 'block';
-    let randomNumbers = generateNumbers();
-    displayNumbers(randomNumbers);
+    displayNumbers();
```

Comment:
I initially changed the display numbers function to take the return from the generateNumbers as a parameter so that we could work with the random array generated outside of the displayNumbers function. do you think having all our logic within the displayNumbers function is better?

Reply:
keep in mind now if we need to restart player on fail, the displayNumbers function can't simply be rerun.